src/java.base/share/classes/jdk/internal/lang/stable/StableFieldUpdater.java

```diff
@@ -152,6 +156,24 @@ private StableFieldUpdater() {}
 
     private static final Unsafe UNSAFE = Unsafe.getUnsafe();
 
+    public static <T> ToIntFunction<T> ofInt(VarHandle accessor,
+                                             MethodHandle underlying,
+                                             int zeroReplacement) {
+        Objects.requireNonNull(accessor);
+        Objects.requireNonNull(underlying);
+
+        if (accessor.varType() != int.class) {
+            throw new IllegalArgumentException("Illegal accessor: " + accessor);
+        }
+        if (underlying.type().returnType() != int.class || underlying.type().parameterCount() != 1) {
+            throw new IllegalArgumentException("Illegal underlying function: " + underlying);
+        }
+        if (!underlying.type().parameterType(0).equals(Object.class)) {
+            underlying = underlying.asType(underlying.type().changeParameterType(0, Object.class));
+        }
```

Comment:
```suggestion
        var type = underlying.type();
        if (type.returnType() != int.class || type.parameterCount() != 1) {
            throw new IllegalArgumentException("Illegal underlying function: " + underlying);
        }
        if (!type.parameterType(0).equals(Object.class)) {
            underlying = underlying.asType(type.changeParameterType(0, Object.class));
        }
```
underlying.type() is used 4 times, local variables should be used